src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs

```diff
@@ -34,43 +34,77 @@ internal class BindingRedirectGenerator
 
         private readonly SHA1 _sha1 = SHA1.Create();
 
-        public XDocument Generate(IEnumerable<LibraryExport> dependencies)
+        public XDocument Generate(IEnumerable<LibraryExport> dependencies, XDocument document)
         {
-            var redirects = CollectRedirects(dependencies);
-            
+			var redirects = CollectRedirects(dependencies);
+
             if (!redirects.Any())
             {
                 // No redirects required
                 return null;
             }
+            document = document ?? new XDocument();
+
+            var configuration = GetOrAddElement(document, ConfigurationElementName);
+            var runtime = GetOrAddElement(configuration, RuntimeElementName);
+            var assemblyBindings = GetOrAddElement(runtime, AssemblyBindingElementName);
+
+            foreach (var redirect in redirects)
+            {
+                AddDependentAssembly(redirect, assemblyBindings);
+            }
 
-            var document = new XDocument(
-                new XElement(ConfigurationElementName,
-                    new XElement(RuntimeElementName,
-                        new XElement(AssemblyBindingElementName,
-                            redirects.Select(GetDependentAssembly)
-                            )
-                        )
-                    )
-                );
             return document;
         }
 
-        private XElement GetDependentAssembly(AssemblyRedirect redirect)
+        private void AddDependentAssembly(AssemblyRedirect redirect, XElement assemblyBindings)
```

Comment:
We should clear out the existing binding redirects and just write new ones. I'm not sure we care about merging them.